src/Engine/GameEvents/BattleEvent.java

```diff
@@ -71,4 +72,10 @@ public void performEvent(GameMap gameMap)
       defender.CO.units.remove(defender);
     }
   }
+  
+  @Override // there's no known way for this to fail after the GameAction is constructed
+  public boolean shouldPreempt(MapMaster gameMap )
```

Comment:
BattleEvent is always preceded by a MoveEvent.
If the MoveEvent succeeds, the BattleEvent *will* be valid since the BattleEvent can't be constructed with an invalid target.
The referenced failure is for the BattleEvent as a whole, not that function.

As for that refactor... you already do that with resupply, so go for it.